addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/handler/EnergenieSubdevicesHandler.java

```diff
@@ -0,0 +1,655 @@
+        String label = getThing().getLabel();
+        JsonObject updateResponse = gatewayHandler.updateSubdevice(this.energenieID, label);
+        if (updateResponse == null) {
+            logger.warn("Failed to update label of newly paired subdevice {} to {}", this.energenieID, label);
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        scheduleRegularUpdate(updateInterval);
+    }
+
+    private void scheduleRegularUpdate(long interval) {
+        if (updateTask != null) {
+            this.updateTask.cancel(true);
+            this.updateTask = null;
+        }
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    if (getThing().getStatus() == ThingStatus.ONLINE) {
+                        JsonObject data = gatewayHandler.getSubdeviceData(energenieID);
+                        if (data != null) {
+                            // Update the channels
+                            List<Channel> channels = getThing().getChannels();
+                            for (Channel channel : channels) {
+                                ChannelUID uid = channel.getUID();
+                                if (isLinked(uid.getId())) {
+                                    updateThingState(data, uid);
+                                }
+                            }
+
+                            // Update the label if changed
+                            JsonElement newLabelElement = data.get(DeviceConstants.DEVICE_LABEL_KEY);
+                            if (newLabelElement != null) {
+                                String newLabel = newLabelElement.getAsString();
+                                String currentLabel = getThing().getLabel();
+                                if (!newLabel.equals(currentLabel)) {
+                                    getThing().setLabel(newLabel);
+                                }
+                            }
+                        } else {
+                            logger.info(
+                                    "Regular update will be stopped! Device may be deleted from the Mi|Home Server.");
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Subdevice " + energenieID + " is missing. It might be deleted!");
+                        }
+
+                    }
+                } catch (Exception e) {
+                    logger.error(
+                            "Exception occurred during execution of update task for device with ID {}. Subsequent executions will be cancelled : ",
+                            energenieID, e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+                    // The Exception will be caught by the ScheduleExecutorService and it will stop subsequent
+                    // executions
+                    throw new RuntimeException(e);
```

Comment:
Maybe an IllegalStateException would fit here, but please pass some context we might start logging these exception very soon :-)